subsys/bluetooth/audio/has.c

```diff
@@ -73,34 +89,55 @@ static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *at
 /* Hearing Access Service GATT Attributes */
 static struct bt_gatt_attr has_attrs[] = {
 	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS),
+
+#if defined(CONFIG_BT_HAS_FEATURES_NOTIFIABLE)
+	BT_AUDIO_CHRC(BT_UUID_HAS_HEARING_AID_FEATURES,
+		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
+		      BT_GATT_PERM_READ_ENCRYPT,
+		      read_features, NULL, NULL),
+	BT_AUDIO_CCC(features_cfg_changed),
+#else
 	BT_AUDIO_CHRC(BT_UUID_HAS_HEARING_AID_FEATURES,
 		      BT_GATT_CHRC_READ,
 		      BT_GATT_PERM_READ_ENCRYPT,
 		      read_features, NULL, NULL),
+#endif /* CONFIG_BT_HAS_FEATURES_NOTIFIABLE */
 #if defined(CONFIG_BT_HAS_PRESET_SUPPORT)
 	BT_AUDIO_CHRC(BT_UUID_HAS_PRESET_CONTROL_POINT,
-#if defined(CONFIG_BT_EATT)
+#if defined(BT_HAS_PRESET_CONTROL_POINT_NOTIFIABLE)
 		      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE | BT_GATT_CHRC_NOTIFY,
-#else
+#else /* CONFIG_BT_HAS_PRESET_SUPPORT */
 		      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE,
 #endif /* CONFIG_BT_EATT */
 		      BT_GATT_PERM_WRITE_ENCRYPT,
 		      NULL, write_control_point, NULL),
-	BT_AUDIO_CCC(ccc_cfg_changed),
+	BT_AUDIO_CCC(preset_cp_cfg_changed),
+#if defined(CONFIG_BT_HAS_ACTIVE_PRESET_INDEX)
 	BT_AUDIO_CHRC(BT_UUID_HAS_ACTIVE_PRESET_INDEX,
 		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
 		      BT_GATT_PERM_READ_ENCRYPT,
 		      read_active_preset_index, NULL, NULL),
-	BT_AUDIO_CCC(ccc_cfg_changed),
+	BT_AUDIO_CCC(active_preset_index_cfg_changed),
+#endif /* CONFIG_BT_HAS_ACTIVE_PRESET_INDEX */
 #endif /* CONFIG_BT_HAS_PRESET_SUPPORT */
 };
```

Comment:
We could consider making this service declaration a bit simpler to read by moving the individual characteristics to individual `#defines` now that there are so many conditionals here. See e.g. how it's done in `tbs.c`